### ISSUE TITLE
Image paths for PDF not calculated correctly when map in sub-directory

### ISSUE THREAD

**Issue by strgaltentf** [dita-ot/dita-ot#2680]
In the attached file structure there are two map files. Map1 is at the root point of the tree; Map2 is in a sub-directory, "_Subdir_version", under the project root.

I built** a pdf using release 2.4.6, using these, then that for Map1 builds correctly. That for Map2 gives the following errors:

> [fop] SEVERE: Image not found. URI: file:/D:/_Dita_ImagePathBug_Document/en_GB/../images/Placeholder.png. (See position 2:-1) 
[fop] SEVERE: Image not found. URI: file:/D:/_Dita_ImagePathBug_Document/en_GB/../images/Placeholder.png. (See position 3:-1) 

The path should have been `D:/_Dita_ImagePathBug_Document/en_GB/_Subdir_version/../images/Placeholder.png`

The problem appears to lie in the file `plugins\org.dita.pdf2\xsl\fo\commons.xsl`, in the template:
```xml     
<xsl:template match="*[contains(@class,' topic/image ')]">
```
Here there are two references similar to the below (lines 1890, 1906 in release 2.4.6):
```xml
<xsl:with-param name="href" select="if (@scope = 'external' or opentopic-func:isAbsolute(@href)) then @href else concat($input.dir.url, @href)"/>
```     
In the `concat` the code implies the first term, `$input.dir.url` is the user-specified input dir (that is, in the example: `D:/_Dita_ImagePathBug_Document/en_GB/_Subdir_version`).  But it's not. It actually seems to be the `root/base` dir needed to include all files for the project. In the example it is  `D:/_Dita_ImagePathBug_Document/en_GB/`.

So, for the relative links to work correctly, the concat also needs to include the relative path from the root dir to the directory holding the map being built:
`concat($root.dir.url, $reldir.to.input.dir.url, @href)`.

I did a temporary work-around for myself, calculating the relative path part in the ant build file and passing it as a further parameter to the xslt; this has cured the problem. I'm not going to suggest it because I'm sure there's a neater way of doing it just within commons.xsl. The variable "$input.dir.url" should probably be renamed too, to match what it actually represents.

**The build I used for the failed build had the paramaters:

> -f pdf2 -Dargs.chapter.layout=BASIC -Dargs.figurelink.style=NUMBER -Dgenerate.copy.outer=3 -Dargs.draft=yes -Dclean.temp=no -temp D:\build\temp -v -i D:\_Dita_ImagePathBug_Document\en_GB\_Subdir_version\Map2.ditamap -o D:\build\out-pdf\_Dita_ImagePathBug_Document\en_GB\_Subdir_version

[_Dita_ImagePathBug_Document.zip](https://github.com/dita-ot/dita-ot/files/985799/_Dita_ImagePathBug_Document.zip)

**Comment by robander** [comment]
This sounds oddly similar to one I just submitted a fix for -- #2550 -- except that it was a problem in the `copy-images` task, which isn't at fault here. But the root cause was much the same - it was using a variable that was set to the common root/base dir, instead of a variable set to the map's directory. In that case, the variable I needed already existed as property `_dita.map.output.dir` - I'm guessing that wouldn't make sense for all uses of the `$input.dir.url` variable, but it might be something you can use without having to calculate a value.

**Comment by bpopp** [comment]
@robander That dita.map.output.dir variable is a fairly commonly used variable on the ANT side, but it doesn't currently appear to be getting sent to the XSL. I'm running into the same issue as OP. It seems like we should be able to set args.input.dir to the parent directory of the source file, but this doesn't seem to be making it to 'input.dir.url' variable. @strgaltentf, could you post the change you made to work around the issue?
